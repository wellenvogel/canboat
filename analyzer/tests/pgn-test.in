#
# A list of PGNs to be processed, in order of PGN.
#
# Some are artificially changed to have non-null fields. These
# are recognisable by timestamp date 1970-01-01.
#
2011-04-25-06:25:03.603,3,129029,36,255,43,e6,f1,3a,80,9c,c6,0d,00,12,38,aa,49,eb,51,07,00,0c,44,95,fb,15,b8,00,40,e1,33,00,00,00,00,00,13,fc,09,5a,00,8c,00,ff,ff,ff,7f,00
1970-01-01T00:00:00.000Z,3,127513,61,255,8,00,c1,11,14,00,02,01,62
2022-09-10T12:10:16.614Z,6,60928,5,255,8,fb,9b,70,22,00,9b,50,c0
2022-09-10T12:10:16.812Z,6,60928,35,255,8,19,e8,e4,10,00,82,78,c0
2016-04-09T16:41:39.628Z,2,127489,16,255,26,00,2f,06,ff,ff,e3,73,65,05,ff,7f,72,10,00,00,ff,ff,ff,ff,ff,06,00,00,00,7f,7f
1970-01-01T16:41:39.628Z,2,127489,16,255,26,00,2f,06,10,20,e3,73,65,05,65,04,72,10,00,00,10,20,30,40,ff,06,00,ff,00,30,18
2022-09-10T12:10:33.618Z,6,129540,23,255,216,04,ff,12,03,73,23,fe,3b,e4,0c,00,00,00,00,f2,57,0b,20,37,82,e4,0c,00,00,00,00,f2,04,73,32,2c,62,80,0c,00,00,00,00,f2,48,ad,2d,96,1a,80,0c,00,00,00,00,f2,49,51,0e,22,24,80,0c,00,00,00,00,f2,31,c5,13,66,7b,1c,0c,00,00,00,00,f2,58,45,2a,fc,c2,1c,0c,00,00,00,00,f2,06,f3,1b,42,cf,b8,0b,00,00,00,00,f2,51,45,0c,aa,e1,b8,0b,00,00,00,00,f2,09,5c,1f,2b,8f,54,0b,00,00,00,00,f2,11,b9,11,71,9b,54,0b,00,00,00,00,f2,13,8b,18,e5,af,54,0b,00,00,00,00,f2,47,45,0c,dc,26,f0,0a,00,00,00,00,f2,41,d0,24,88,ae,8c,0a,00,00,00,00,f2,0b,22,06,b6,d4,28,0a,00,00,00,00,f2,01,a2,0d,38,64,fc,08,00,00,00,00,f2,19,ba,02,64,f3,98,08,00,00,00,00,f2,4a,7f,07,d0,42,34,08,00,00
2020-08-22-13:52:57.591,7,126993,36,255,8,01,00,24,ff,ff,ff,ff,ff
2011-04-25-10:16:40.505,3,126992,36,255,8,10,f0,f1,3a,21,47,0f,16
2021-01-30-20:43:21.684,6,126998,1,255,19,07,01,68,65,6C,6C,6F,0c,00,77,00,F3,00,72,00,6C,00,64,00
2020-04-19T00:35:55.571Z,2,126208,0,67,21,01,16,f0,01,ff,01,02,0e,01,59,44,3a,56,4f,4c,55,4d,45,20,36,30
2021-07-29T10:18:31.758Z,6,126208,36,0,7,02,82,ff,00,10,02,00
2021-07-29T10:18:31.758Z,6,126208,36,0,11,03,02,fd,01,00,01,02,04,02,02,03
2022-10-11T11:47:22Z,3,126464,127,255,7,01,04,ff,01,11,fb,01
2022-11-14T01:47:30.890Z,2,127251,14,255,8,ff,51,bf,ff,ff,ff,ff,ff
2022-09-10T12:07:29.542Z,4,129039,23,255,27,12,8a,e4,8d,0e,b4,c4,2a,03,22,d7,88,1f,77,09,75,b4,00,f8,08,00,ff,ff,00,f0,fe,ff
#SHOWBUFFERS
