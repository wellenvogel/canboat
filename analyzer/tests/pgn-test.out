2011-04-25-06:25:03.603 3  36 255 129029 GNSS Position Data:  SID = 230 (bytes = "E6"); Date = 2011.04.25 (bytes = "F1 3A"); Time = 06:25:12 (bytes = "80 9C C6 0D"); Latitude = 52.7461333 (bytes = "00 12 38 AA 49 EB 51 07"); Longitude =  5.1815566 (bytes = "00 0C 44 95 FB 15 B8 00"); Altitude = 3.400000 m (bytes = "40 E1 33 00 00 00 00 00"); GNSS type = GPS+SBAS/WAAS (bytes = "03", bits = "0011"); Method = GNSS fix (bytes = "10", bits = "0001"); Integrity = No integrity checking (bytes = "00", bits = "00"); Number of SVs = 9 (bytes = "09"); HDOP = 0.90 (bytes = "5A 00"); PDOP = 1.40 (bytes = "8C 00"); Geoidal Separation = Unknown (bytes = "FF FF FF 7F"); Reference Stations = 0 (bytes = "00")
1970-01-01T00:00:00.000Z 3  61 255 127513 Battery Configuration Status:  Instance = 0 (bytes = "00"); Battery Type = Gel (bytes = "01", bits = "0001"); Supports Equalization = No (bytes = "00", bits = "00"); Nominal Voltage = 12V (bytes = "01", bits = "0001"); Chemistry = Li (bytes = "10", bits = "0001"); Capacity = 20 Ah (bytes = "14 00"); Temperature Coefficient = 2 % (bytes = "02"); Peukert Exponent = 1.002 (bytes = "01"); Charge Efficiency Factor = 98 % (bytes = "62")
2022-09-10T12:10:16.614Z 6   5 255  60928 ISO Address Claim:  Unique Number = 1088507 (bytes = "FB 9B 10", bits = "111101111110111111011"); Manufacturer Code = Navico (bytes = "60 22", bits = "00100010011"); Device Instance Lower = 0 (bytes = "00", bits = "000"); Device Instance Upper = 0 (bytes = "00", bits = "00000"); Device Function = Rudder (bytes = "9B"); Device Class = Steering and Control surfaces (bytes = "50", bits = "0101000"); System Instance = 0 (bytes = "00", bits = "0000"); Industry Group = Marine (bytes = "40", bits = "100"); Arbitrary address capable = 1 (bytes = "80", bits = "1")
2022-09-10T12:10:16.812Z 6  35 255  60928 ISO Address Claim:  Unique Number = 321561 (bytes = "19 E8 04", bits = "001100000110000011001"); Manufacturer Code = Airmar (bytes = "E0 10", bits = "01110000111"); Device Instance Lower = 0 (bytes = "00", bits = "000"); Device Instance Upper = 0 (bytes = "00", bits = "00000"); Device Function = Bottom Depth (bytes = "82"); Device Class = Navigation (bytes = "78", bits = "0111100"); System Instance = 0 (bytes = "00", bits = "0000"); Industry Group = Marine (bytes = "40", bits = "100"); Arbitrary address capable = 1 (bytes = "80", bits = "1")
2016-04-09T16:41:39.628Z 2  16 255 127489 Engine Parameters, Dynamic:  Instance = Single Engine or Dual Engine Port (bytes = "00"); Oil pressure = 1.583 bar (bytes = "2F 06"); Oil temperature = Unknown (bytes = "FF FF"); Temperature = 23.52 C (bytes = "E3 73"); Alternator Potential = 13.81 V (bytes = "65 05"); Fuel Rate = Unknown (bytes = "FF 7F"); Total Engine hours = 01:10:10 (bytes = "72 10 00 00"); Coolant Pressure = Unknown (bytes = "FF FF"); Fuel Pressure = Unknown (bytes = "FF FF"); Discrete Status 1 = Over Temperature,Low Oil Pressure (bytes = "06 00"); Discrete Status 2 = None (bytes = "00 00"); Engine Load = Unknown (bytes = "7F"); Engine Torque = Unknown (bytes = "7F")
1970-01-01T16:41:39.628Z 2  16 255 127489 Engine Parameters, Dynamic:  Instance = Single Engine or Dual Engine Port (bytes = "00"); Oil pressure = 1.583 bar (bytes = "2F 06"); Oil temperature = 547.65 C (bytes = "10 20"); Temperature = 23.52 C (bytes = "E3 73"); Alternator Potential = 13.81 V (bytes = "65 05"); Fuel Rate = 112.5 L/h (bytes = "65 04"); Total Engine hours = 01:10:10 (bytes = "72 10 00 00"); Coolant Pressure = 8.208 bar (bytes = "10 20"); Fuel Pressure = 164.320 bar (bytes = "30 40"); Discrete Status 1 = Over Temperature,Low Oil Pressure (bytes = "06 00"); Discrete Status 2 = Warning Level 1,Warning Level 2,Power Reduction,Maintenance Needed,Engine Comm Error,Sub or Secondary Throttle,Neutral Start Protect,Engine Shutting Down (bytes = "FF 00"); Engine Load = 48 % (bytes = "30"); Engine Torque = 24 % (bytes = "18")
2022-09-10T12:10:33.618Z 6  23 255 129540 GNSS Sats in View:  SID = 4 (bytes = "04"); Range Residual Mode = Unknown (bytes = "03", bits = "11"); Sats in View = 18 (bytes = "12"); PRN 1 = 3 (bytes = "03"); Elevation 1 = 52.0 deg (bytes = "73 23"); Azimuth 1 = 88.0 deg (bytes = "FE 3B"); SNR 1 = 33.00 dB (bytes = "E4 0C"); Range residuals 1 = 0 (bytes = "00 00 00 00"); Status 1 = Used (bytes = "02", bits = "0010"); PRN 2 = 87 (bytes = "57"); Elevation 2 = 47.0 deg (bytes = "0B 20"); Azimuth 2 = 191.0 deg (bytes = "37 82"); SNR 2 = 33.00 dB (bytes = "E4 0C"); Range residuals 2 = 0 (bytes = "00 00 00 00"); Status 2 = Used (bytes = "02", bits = "0010"); PRN 3 = 4 (bytes = "04"); Elevation 3 = 74.0 deg (bytes = "73 32"); Azimuth 3 = 144.0 deg (bytes = "2C 62"); SNR 3 = 32.00 dB (bytes = "80 0C"); Range residuals 3 = 0 (bytes = "00 00 00 00"); Status 3 = Used (bytes = "02", bits = "0010"); PRN 4 = 72 (bytes = "48"); Elevation 4 = 67.0 deg (bytes = "AD 2D"); Azimuth 4 = 39.0 deg (bytes = "96 1A"); SNR 4 = 32.00 dB (bytes = "80 0C"); Range residuals 4 = 0 (bytes = "00 00 00 00"); Status 4 = Used (bytes = "02", bits = "0010"); PRN 5 = 73 (bytes = "49"); Elevation 5 = 21.0 deg (bytes = "51 0E"); Azimuth 5 = 53.0 deg (bytes = "22 24"); SNR 5 = 32.00 dB (bytes = "80 0C"); Range residuals 5 = 0 (bytes = "00 00 00 00"); Status 5 = Used (bytes = "02", bits = "0010"); PRN 6 = 49 (bytes = "31"); Elevation 6 = 29.0 deg (bytes = "C5 13"); Azimuth 6 = 181.0 deg (bytes = "66 7B"); SNR 6 = 31.00 dB (bytes = "1C 0C"); Range residuals 6 = 0 (bytes = "00 00 00 00"); Status 6 = Used (bytes = "02", bits = "0010"); PRN 7 = 88 (bytes = "58"); Elevation 7 = 62.0 deg (bytes = "45 2A"); Azimuth 7 = 286.0 deg (bytes = "FC C2"); SNR 7 = 31.00 dB (bytes = "1C 0C"); Range residuals 7 = 0 (bytes = "00 00 00 00"); Status 7 = Used (bytes = "02", bits = "0010"); PRN 8 = 6 (bytes = "06"); Elevation 8 = 41.0 deg (bytes = "F3 1B"); Azimuth 8 = 304.0 deg (bytes = "42 CF"); SNR 8 = 30.00 dB (bytes = "B8 0B"); Range residuals 8 = 0 (bytes = "00 00 00 00"); Status 8 = Used (bytes = "02", bits = "0010"); PRN 9 = 81 (bytes = "51"); Elevation 9 = 18.0 deg (bytes = "45 0C"); Azimuth 9 = 331.0 deg (bytes = "AA E1"); SNR 9 = 30.00 dB (bytes = "B8 0B"); Range residuals 9 = 0 (bytes = "00 00 00 00"); Status 9 = Used (bytes = "02", bits = "0010"); PRN 10 = 9 (bytes = "09"); Elevation 10 = 46.0 deg (bytes = "5C 1F"); Azimuth 10 = 210.0 deg (bytes = "2B 8F"); SNR 10 = 29.00 dB (bytes = "54 0B"); Range residuals 10 = 0 (bytes = "00 00 00 00"); Status 10 = Used (bytes = "02", bits = "0010"); PRN 11 = 17 (bytes = "11"); Elevation 11 = 26.0 deg (bytes = "B9 11"); Azimuth 11 = 228.0 deg (bytes = "71 9B"); SNR 11 = 29.00 dB (bytes = "54 0B"); Range residuals 11 = 0 (bytes = "00 00 00 00"); Status 11 = Used (bytes = "02", bits = "0010"); PRN 12 = 19 (bytes = "13"); Elevation 12 = 36.0 deg (bytes = "8B 18"); Azimuth 12 = 258.0 deg (bytes = "E5 AF"); SNR 12 = 29.00 dB (bytes = "54 0B"); Range residuals 12 = 0 (bytes = "00 00 00 00"); Status 12 = Used (bytes = "02", bits = "0010"); PRN 13 = 71 (bytes = "47"); Elevation 13 = 18.0 deg (bytes = "45 0C"); Azimuth 13 = 57.0 deg (bytes = "DC 26"); SNR 13 = 28.00 dB (bytes = "F0 0A"); Range residuals 13 = 0 (bytes = "00 00 00 00"); Status 13 = Used (bytes = "02", bits = "0010"); PRN 14 = 65 (bytes = "41"); Elevation 14 = 54.0 deg (bytes = "D0 24"); Azimuth 14 = 256.0 deg (bytes = "88 AE"); SNR 14 = 27.00 dB (bytes = "8C 0A"); Range residuals 14 = 0 (bytes = "00 00 00 00"); Status 14 = Used (bytes = "02", bits = "0010"); PRN 15 = 11 (bytes = "0B"); Elevation 15 = 9.0 deg (bytes = "22 06"); Azimuth 15 = 312.0 deg (bytes = "B6 D4"); SNR 15 = 26.00 dB (bytes = "28 0A"); Range residuals 15 = 0 (bytes = "00 00 00 00"); Status 15 = Used (bytes = "02", bits = "0010"); PRN 16 = 1 (bytes = "01"); Elevation 16 = 20.0 deg (bytes = "A2 0D"); Azimuth 16 = 147.0 deg (bytes = "38 64"); SNR 16 = 23.00 dB (bytes = "FC 08"); Range residuals 16 = 0 (bytes = "00 00 00 00"); Status 16 = Used (bytes = "02", bits = "0010"); PRN 17 = 25 (bytes = "19"); Elevation 17 = 4.0 deg (bytes = "BA 02"); Azimuth 17 = 357.0 deg (bytes = "64 F3"); SNR 17 = 22.00 dB (bytes = "98 08"); Range residuals 17 = 0 (bytes = "00 00 00 00"); Status 17 = Used (bytes = "02", bits = "0010"); PRN 18 = 74 (bytes = "4A"); Elevation 18 = 11.0 deg (bytes = "7F 07"); Azimuth 18 = 98.0 deg (bytes = "D0 42"); SNR 18 = 21.00 dB (bytes = "34 08"); Range residuals 18 = 0 (bytes = "00 00")
2020-08-22-13:52:57.591 7  36 255 126993 Heartbeat:  Data transmit offset = 00:00:00.001 (bytes = "01 00"); Sequence Counter = 36 (bytes = "24"); Controller 1 State = Unknown (bytes = "03", bits = "11"); Controller 2 State = Unknown (bytes = "0C", bits = "11"); Equipment Status = Unknown (bytes = "30", bits = "11")
2011-04-25-10:16:40.505 3  36 255 126992 System Time:  SID = 16 (bytes = "10"); Source = GPS (bytes = "00", bits = "0000"); Date = 2011.04.25 (bytes = "F1 3A"); Time = 10:16:50.0001 (bytes = "21 47 0F 16")
2021-01-30-20:43:21.684 6   1 255 126998 Configuration Information:  Installation Description #1 = hello (bytes = "07 01 68 65 6C 6C 6F"); Installation Description #2 = w√≥rld (bytes = "0C 00 77 00 F3 00 72 00 6C 00 64 00")
2020-04-19T00:35:55.571Z 2   0  67 126208 NMEA - Command group function:  Function Code = Command (bytes = "01"); PGN = 126998 (bytes = "16 F0 01"); Priority = Unknown (bytes = "0F", bits = "1111"); Number of Parameters = 1 (bytes = "01"); Parameter 1 = 2 (bytes = "02"); Value 1 = YD:VOLUME 60 (bytes = "0E 01 59 44 3A 56 4F 4C 55 4D 45 20 36 30")
2021-07-29T10:18:31.758Z 6  36   0 126208 NMEA - Acknowledge group function:  Function Code = Acknowledge (bytes = "02"); PGN = 65410 (bytes = "82 FF 00"); PGN error code = Acknowledge (bytes = "00", bits = "0000"); Transmission interval/Priority error code = Transmit Interval/Priority not supported (bytes = "10", bits = "0001"); Number of Parameters = 2 (bytes = "02"); Parameter 1 = Acknowledge (bytes = "00", bits = "0000"); Parameter 2 = Acknowledge (bytes = "00", bits = "0000")
2021-07-29T10:18:31.758Z 6  36   0 126208 NMEA - Read Fields group function:  Function Code = Read Fields (bytes = "03"); PGN = 130306 (bytes = "02 FD 01"); Unique ID = 0 (bytes = "00"); Number of Selection Pairs = 1 (bytes = "01"); Number of Parameters = 2 (bytes = "02"); Selection Parameter 1 = 4 (bytes = "04"); Selection Value 1 = Apparent (bytes = "02", bits = "010"); Parameter 1 = 2 (bytes = "02"); Parameter 2 = 3 (bytes = "03")
2022-10-11T11:47:22Z 3 127 255 126464 PGN List (Transmit and Receive):  Function Code = Receive PGN list (bytes = "01"); PGN 1 = 130820 (bytes = "04 FF 01"); PGN 2 = 129809 (bytes = "11 FB 01")
2022-11-14T01:47:30.890Z 2  14 255 127251 Rate of Turn:  SID = Unknown (bytes = "FF"); Rate = -0.029649 deg/s (bytes = "51 BF FF FF")
2022-09-10T12:07:29.542Z 4  23 255 129039 AIS Class B Position Report:  Message ID = Standard Class B position report (bytes = "12", bits = "010010"); Repeat Indicator = Initial (bytes = "00", bits = "00"); User ID = "244180106" (bytes = "8A E4 8D 0E"); Longitude =  5.3134516 (bytes = "B4 C4 2A 03"); Latitude = 52.9061666 (bytes = "22 D7 88 1F"); Position Accuracy = High (bytes = "01", bits = "1"); RAIM = in use (bytes = "02", bits = "1"); Time Stamp = 29 (bytes = "74", bits = "011101"); COG = 171.7 deg (bytes = "09 75"); SOG = 1.80 m/s (bytes = "B4 00"); Communication State = F8 08 00 (bytes = "F8 08 00", bits = "1100111110011111000"); AIS Transceiver information = Channel A VDL reception (bytes = "00", bits = "00000"); Heading = Unknown (bytes = "FF FF"); Unit type = SOTDMA (bytes = "00", bits = "0"); Integrated Display = No (bytes = "00", bits = "0"); DSC = Yes (bytes = "10", bits = "1"); Band = Entire marine band (bytes = "20", bits = "1"); Can handle Msg 22 = Yes (bytes = "40", bits = "1"); AIS mode = Assigned (bytes = "80", bits = "1"); AIS communication state = SOTDMA (bytes = "00", bits = "0")
1970-01-01T00:00:00.000Z 1   1 255 129799 Radio Frequency/Mode/Power:  Rx Frequency = 156800000 Hz (bytes = "00 42 EF 00"); Tx Frequency = 156800000 Hz (bytes = "00 42 EF 00"); Radio Channel = 900016 (bytes = "39 30 30 30 31 36"); Tx Power = 25 W (bytes = "19"); Mode = 0 (bytes = "00 00"); Channel Bandwidth = 25000 Hz (bytes = "A8 61")
